content/blog/2025/oddcontrast-guide.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Make the Web a More a Colorful Place!
+sub: A guide to using new color spaces & formats with OddContrast
+author: oddbird
+date: 2025-06-05
+tags:
+  - Article
+  - Accessibility
+  - Color
+  - Design
+  - OddContrast
+  - OddTools
+image:
+  src: blog/2025/oddcontrast-guide.jpg
+  alt: >
+    OddContrast displays P3 gamut range
+    in Oklch color format
+summary: |
+  OddBird's color tool not only checks contrast ratios,
+  but supports the new CSS color formats and spaces.
+---
+
+{% import 'embed.macros.njk' as embed %}
+
+For years designers and developers were limited to colors in the sRGB colors
+space, using formats like Hexadecimal, RGB, and HSL. As display technology
+progressed, so too has CSS, and we have access to additional color spaces and
+wider gamuts. These advances led us to build OddContrast, a color editing and
+testing tool that handles our new world of modern color formats.
+
+## The flexibility of OddContrast
+
+Although it looks simple on the surface, OddContrast has many built-in
+conveniences when working with color.
+
+The default color format is display-p3 but you can select from the following
+options:
+- Hex
+- HSL
+- Lab
+- LCH
+- Oklab
+- Oklch
+- P3
+- sRGB
+
+{{ embed.figure(
+  data=[{
+    img: 'blog/2025/oddcontrast-formats.jpg',
+    alt: 'p3 is selected in the color format select field'
+  }]
+) }}
+
+Regardless of chosen color display format,
+you can use any valid CSS notation to enter foreground and background colors.
+For instance, you can use decimals in `oklch(.44 0.1 295)`
+even though the default is set to display the lightness value as a `%`.
+You can even used named colors! Here’s `honeydew` as the foreground color.
+
+{{ embed.figure(
+  data=[{
+    img: 'blog/2025/oddcontrast-named-colors-cropped.jpg',
+    alt: 'honeydew named colors is used as the foreground color'
+  }]
+) }}
+
+While not valid CSS notation, OddContrast also accepts 6 and 8-digit Hex
+colors without the `#`.
+This flexibility is helpful when coming from design tools as they
+often will not include the full color notation.
+
+## Workflow enhancements
+
+To streamline your color-editing tasks, OddContrast includes several helpful
+features. For starters, you can edit a single component of a color by using the
+visual sliders. These change depending on the color format you've selected. For
+instance, if Oklab is selected, the first slider controls the lightness (L).
+The next two sliders show options from green to red (A) and blue to yellow (B).
+The last slider sets the Alpha value and is consistent across all color formats.
+Additionally, you can use your keyboard arrow keys to nudge the color values
+incrementally when focused.
+
+Need to swap the foreground and background colors? No problem, select the left
+right arrow icon button or you can drag and drop the color swatches themselves.
```

Comment:
```suggestion
Need to swap the foreground and background colors? No problem -- select the
left-right arrow icon button, or drag-and-drop the color swatches themselves.
```

Reply:
I'm fine either way, just thought it was easier to read/parse as `left-right` 🤷

Reply:
updated in 613995ff159685af68a52ea345394d3d1f006710